x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.tsx

```diff
@@ -0,0 +1,415 @@
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
+import { useSourcererDataView } from '../../../../common/containers/sourcerer';
+import { getColumnHeader } from '../body/column_headers/helpers';
+import { timelineBodySelector } from '../body/selectors';
+import { StyledPageContentWrapper, StyledMainEuiPanel, StyledSplitFlexItem } from './styles';
+import { DRAG_DROP_FIELD } from './data_table/translations';
+import { TimelineResizableLayout } from './resizable_layout';
+import TimelineDataTable from './data_table';
+import { timelineDefaults } from '../../../store/defaults';
+import { timelineActions } from '../../../store';
+import { getFieldsListCreationOptions } from './get_fields_list_creation_options';
+
+const TimelineBodyContainer = styled.div.attrs(({ className = '' }) => ({
+  className: `${className}`,
+}))`
+  width: 100%;
+  height: 100%;
+`;
+
+const DataGridMemoized = React.memo(TimelineDataTable);
+
+const DROP_PROPS = {
+  value: {
+    id: 'dscDropZoneTable',
+    humanData: {
+      label: DRAG_DROP_FIELD,
+    },
+  },
+  order: [1, 0, 0, 0],
+  types: ['field_add'] as DropType[],
+};
+
+const SidebarPanelFlexGroup = styled(EuiFlexGroup)`
+  height: 100%;
+
+  .unifiedFieldListSidebar {
+    padding-bottom: ${(props) => (props.theme as EuiTheme).eui.euiSizeS};
+    padding-left: 0px;
+    border-top: 1px solid ${(props) => (props.theme as EuiTheme).eui.euiColorLightShade};
+
+    .unifiedFieldListSidebar__group {
+      .euiFlexItem:last-child {
+        /* padding-right: ${(props) => (props.theme as EuiTheme).eui.euiSizeS}; */
+      }
+      .unifiedFieldListSidebar__list {
+        padding-left: 0px;
+      }
+
+      .unifiedFieldListSidebar__addBtn {
+        margin-right: ${(props) => (props.theme as EuiTheme).eui.euiSizeS};
+      }
+    }
+  }
+`;
+
+export const SAMPLE_SIZE_SETTING = 500;
+export const HIDE_FOR_SIZES = ['xs', 's'];
+
+interface Props {
+  columns: ColumnHeaderOptions[];
+  rowRenderers: RowRenderer[];
+  timelineId: string;
+  itemsPerPage: number;
+  itemsPerPageOptions: number[];
+  sort: SortColumnTimeline[];
+  events: TimelineItem[];
+  refetch: inputsModel.Refetch;
+  totalCount: number;
+  onEventClosed: (args: ToggleDetailPanel) => void;
+  expandedDetail: ExpandedDetailTimeline;
+  showExpandedDetails: boolean;
+  onChangePage: OnChangePage;
+  activeTab: TimelineTabs;
+  dataLoadingState: DataLoadingState;
+  updatedAt: number;
+  isTextBasedQuery?: boolean;
+}
+
+export const UnifiedTimelineComponent: React.FC<Props> = ({
+  columns,
+  activeTab,
+  timelineId,
+  itemsPerPage,
+  itemsPerPageOptions,
+  rowRenderers,
+  sort,
+  events,
+  refetch,
+  dataLoadingState,
+  totalCount,
+  onEventClosed,
+  showExpandedDetails,
+  expandedDetail,
+  onChangePage,
+  updatedAt,
+  isTextBasedQuery,
+}) => {
+  const dispatch = useDispatch();
+  const unifiedFieldListContainerRef = useRef<UnifiedFieldListSidebarContainerApi>(null);
+  const renderId = useRef(1);
```

Comment:
why is this needed?

Reply:
@kqualters-elastic  

It is not needed and now removed. Thanks for catching it.